docs/ldml/tr35-personNames.md

```diff
@@ -41,26 +41,924 @@ The LDML specification is divided into the following parts:
+
+In more detail:
+
+A set of input parameters { order=O length=L usage=U formality=F } matches a personName element when:
+
+* The order attribute values contain O or there is no order attribute, and
+* The length attribute values contain L or there is no length attribute, and
+* The usage attribute values contain U or there is no usage attribute, and
+* The formality attribute values contain F or there is no formality attribute
+
+Example for input parameters 
+
+    order=**givenFirst** length=**long** usage=**referring** formality=**formal**
+
+To match a personName, all four attributes in the personName must match (a missing attribute matches any value for that attribute):
+
+| Sample personName attributes                                 | Matches? | Comment |
+| :----------------------------------------------------------- | :------: | :------ |
+| `order=_"givenFirst"_ length=_"long"_ usage=_"referring"_ formality=_"formal"_` | Y | exact match |
+| `length=_"long"_ usage=_"referring"_ formality=_"informal"_` | N | mismatch for formality |
+| `length=_"long"_ formality=_"formal"_`                       | Y | missing usage = all! |
+
+To find the matching personName element, traverse all the personNames in order until the first one is found. This will always terminate since the data is well-formed in CLDR.
+
+### 6.5 <a name="6-5-choose-a-namepattern" href="#6-5-choose-a-namepattern">Choose a namePattern</a>
+
+To format a name, the fields in a namePattern are replaced with the actual PersonName object to be formatted values from the input record. The personName element can contain multiple namePattern elements. We choose one based on the fields in the input PersonName object that are populated: The namePattern that has the most populated fields wins; if more than one have the same number of populated fields, the one with the least unpopulated fields wins. For example:
+
+1. Pattern A has 12 fields total, pattern B has 10 fields total, and pattern C has 8 fields total.
+2. Both patterns A and B can be populated with 7 fields from the input PersonName object, pattern C can be populated with only 3 fields from the input PersonName object.
+3. Pattern C is discarded, because it has the least number of populated name fields.
+4. Out of the remaining patterns A and B, pattern B wins, because it has only 3 unpopulated fields compared to pattern A.
+
+If the “winning” namePattern still has fields that are unpopulated in the PersonName object, we alter the pattern algorithmically as follows:
+
+1. For each empty field from the start of the pattern, that field and all whitespace and literal text between it and the previous field and it and the next field is deleted. (This process stops with the first non-empty field.)
+2. For each empty field from the end of the pattern, that field and all whitespace and literal text between it and the next field and it and the previous field is deleted. (This process stops with the first non-empty field.)
```

Comment:
These two steps need to also mention that if the first field is empty, we also remove any whitespace and literal text at the beginning of the string, and that if the last field is empty, we also remove any whitespace and literal text at the end of the string.  It might be better to reword step 1 as follows:

>If one or more fields at the start of the pattern are empty, all fields, whitespace, and literal text before the first populated field are deleted.

...and to reword step 2 in the same way.

Reply:
That is much cleaner!